### ISSUE TITLE
Export hangs at 6.636% complete

### ISSUE THREAD

**Issue by someotherbob** [OpenShot/openshot-qt#5568]
**Describe the bug**
Export hangs at 6.636% complete

**Steps to reproduce the behavior:**
I have a multi clip timeline that was too long, over 10 minutes.  I changed each clip to a 2x fast forward to reduce the time. Export now hangs.

On each clip I right click and select Time, Fast , Forward, 2x.
Export at 1080p 60fps.
Export hangs at 6.636% complete

**Expected behavior**
I expected the export to complete.

**System Details**
- OpenShot Version: 3.2.0-dev
- libopenshot Version:  0.3.3
- Platform: Windows-10-10.0.19041
- Processor: Intel64 Family 6 Model 122 Stepping 1, GenuineIntel
- Machine: AMD64
- Python version: 3.8.9
- Qt5 version: 5.15.2
- PyQt5 version: 5.15.4
- Qt Detected Languages: ['en-US']
- LANG Environment Variable: 
- LOCALE Environment Variable: 
- Daily Build: Verified issue still exists in daily build: http://github.com/OpenShot/openshot-qt/releases/download/OpenShot-v3.2.0-dev-daily-12555-044bb224-23713d3a-x86_64.exe

**Log Files**
- [openshot-qt.log](https://s3.amazonaws.com/logs.openshot.org/iz74njh5z5qhdoa2jkhzvjvnevr3hirk/openshot-qt.zip) (15 KB)
- libopenshot.log: File too small or empty (not uploaded)

**Exception / Stacktrace**
```No stacktrace found in log files```

**Screenshots** (Optional)
If applicable, add screenshots to help explain your problem. You can include screenshots by
copy/pasting them on GitHub or dragging-and-dropping into the GitHub page. All images are public,
so please don't post screenshots containing personal information.

**Comment by someotherbob** [comment]
Logs
[libopenshot.zip](https://github.com/user-attachments/files/16113223/libopenshot.zip)
[openshot-qt.log](https://github.com/user-attachments/files/16113224/openshot-qt.log)

**Comment by Colorjet3** [comment]
Hello @someotherbob Please provide the following:

1.  The OpenShot build # and/or release date/time.  In Help | About OpenShot, you will see OpenShot-v3.2.0-dev-daily-99999-......  It is the 99999 number that I am asking for.
2.  What type of video files are you working with (.mp4? .mov? etc.)?
3.  What is the approximate duration of videos?
4.  How many clips on the timeline and how many tracks have clips on them?
5.  Are your videos originally recorded in1080p 60fps?  If not, what is the property of your video?  You can find this out by right clicking on one of your video files, selecting Properties, and then the detail tab.
6.  Are you changing the default profile in OpenShot while editing?
7.  Are you changing the profile during Export process only?

I need the answer so I can try and replicate your issue.

**Comment by someotherbob** [comment]
> 1. The OpenShot build # and/or release date/time.  In Help | About OpenShot, you will see OpenShot-v3.2.0-dev-daily-99999-......  It is the 99999 number that I am asking for.

12555

> 2. What type of video files are you working with (.mp4? .mov? etc.)?

mp4 1920x1080 30fps from a Wyze Cam v3 with RTSP FW.

> 3. What is the approximate duration of videos?

Videos are mostly 17 seconds long, created by using the AgentDVR time-lapse function.

> 4. How many clips on the timeline and how many tracks have clips on them?

12 clips all on one track. No audio. Original clips summed up to over 16 minutes, so I applied a 2x fast forward to them all.

> 5. Are your videos originally recorded in1080p 60fps?  If not, what is the property of your video?  You can find this out by right clicking on one of your video files, selecting Properties, and then the detail tab.

mp4 1920x1080 30fps from a Wyze Cam v3 with RTSP FW.

> 6. Are you changing the default profile in OpenShot while editing?

No changes from defaults.

> 7. Are you changing the profile during Export process only?

Export profile is FHD 1080p 60fsp.
 
> I need the answer so I can try and replicate your issue.

**Comment by Colorjet3** [comment]
Thank you @someotherbob for all the information.  Is there a reason you are not keeping the FPS at 30 to match the original?

Things to try:

Option 1.  Instead of Exporting to FHD 1080 60fsp, export at FHD 1080 30fsp.  Does this work?

Option 2.  Open the project and change the Default Profile (while editing) to FHD 1080 60fps.  Now Export.   Does this work?

Option 3.  
a. Start a new project and import your clips into it.
b. Do not make any changes (no edits).
c. Export the project but no changes to the profile from the Export screen.
d. Start a new project and import the exported .mp4 from step c.
e. Drag it onto a track and apply the Fast | forward | 2x.
f.  Export with FHD 1080 60fps.
g. Did this work?

I hope to get to test this over this weekend as well to see what kind of results I get.  I am running Windows 11 Pro on my host pc.  I do have a Windows 10 Pro as a virtual guest but that is not a one for one testing.

**Comment by someotherbob** [comment]
> Option 1. Instead of Exporting to FHD 1080 60fsp, export at FHD 1080 30fsp. Does this work?

Still hangs at 6.636% when exporting project at 1080 30fps. I left the default profile at 720p(?).

> Option 2. Open the project and change the Default Profile (while editing) to FHD 1080 60fps. Now Export. Does this work?

I changed the default profile to 1080p-60.  I restarted Openshot.  Export at 1080p-60 still fails.

> Opetion 3. a. Start a new project and import your clips into it. b. Do not make any changes (no edits). c. Export the project but no changes to the profile from the Export screen. d. Start a new project and import the exported .mp4 from step c. e. Drag it onto a track and apply the Fast | forward | 2x. f. Export with FHD 1080 60fps. g. Did this work?

Life has intervened, I will send results from option 3 on Monday.

**Comment by Colorjet3** [comment]
Thank you again @someotherbob for all the information.  Understood regarding Option 3.  I am very interested in seeing the results from this option.

Two other options I thought of:

Option 4
1.  Open your project.
2.  Click on Export.
4.  Click on the Advanced tab and note "Start Frame" and "End Frame" fields.
5.  "Start Frame:" is always 1.  Let's assume that "End Frame:" is set to 4000.
6.  Change "Start Frame:" to 2000 and leave the End Frame to 4000.
7.  Export.  Did this work?  If yes,
8.  Now Export again but change Start and End to 1000-1999.  Export.  Did this work?

You get the idea.  By changing the Start / End frame you can isolate which clip is possibly the culprit.  You can also work the otherway by changing the Start/End to say 1-200, then 201-400, etc.

Option 5
1.  Take your original files and run them through a converter like VLC, Handbrake, or ShutterEncoder and convert them to .mp4 even though the originals are .mp4.  The conversion process will automatically create the new files with new names.
2.  Now import these new converted files into a new project, perform your edtis, and now export.
3.  Did this work?

We will continue to troubleshoot based on your test results.

By the way, a new daily buid has been released so when you get a chance please upgrade to that one.  It doesn't address this issue but it is best to work with the latest so if we run into other issues we can report it to the lead developer knowing that we are running the latest.

**Comment by someotherbob** [comment]
> Option 3. a. Start a new project and import your clips into it. b. Do not make any changes (no edits). c. Export the project but no changes to the profile from the Export screen. d. Start a new project and import the exported .mp4 from step c. e. Drag it onto a track and apply the Fast | forward | 2x. f. Export with FHD 1080 60fps. g. Did this work?
> 

Added all clips to new project.
Did not fast fwd 2x.
Export completed with no errors.

Resulting mp4 file is not playable in either Windows player, VLC, and cannot be imported into Openshot as a clip.

**Comment by someotherbob** [comment]
> Option 4
> 
> 1. Open your project.
> 2. Click on Export.
> 3. Click on the Advanced tab and note "Start Frame" and "End Frame" fields.
> 4. "Start Frame:" is always 1.  Let's assume that "End Frame:" is set to 4000.
> 5. Change "Start Frame:" to 2000 and leave the End Frame to 4000.
> 6. Export.  Did this work?  If yes,
> 7. Now Export again but change Start and End to 1000-1999.  Export.  Did this work?
> 

Since the hang occurs at less than 10%, I exported the first 10% of the clip.  Frames were 1-32809.  I exported frames 1-3280.
This export hung at 66.39% complete, probably at the exact frame as before.

Export of frames 2000-2300 also hangs.

**Comment by Colorjet3** [comment]
Hmmm....I am surprised at Option 3 result.

Let's see what happens when you try Option 5.

**Comment by someotherbob** [comment]
Since the error is during the first clip, I deleted it and exported the rest.   I got this error.

![Screenshot 2024-07-09 092220](https://github.com/OpenShot/openshot-qt/assets/5491843/75d3eeb9-3ce3-40fb-a040-b3f01ac1903a)

**Comment by someotherbob** [comment]
I put clips 1 & 2 (of 12) on the timeline, now Openshot just closes after starting the export.  See attached logs.

[libopenshot.log](https://github.com/user-attachments/files/16152122/libopenshot.log)
[openshot-qt.log](https://github.com/user-attachments/files/16152124/openshot-qt.log)

**Comment by Colorjet3** [comment]
I see that you are exporting to you D: drive.  How much free disk space do you have available?  The std::badalloc is related to not enough disk space.  Free up as much as possible and try your export again.

**Comment by someotherbob** [comment]
> I see that you are exporting to you D: drive. How much free disk space do you have available? The std::badalloc is related to not enough disk space. Free up as much as possible and try your export again.

D: has 2.62Tbytes free.
C: has 13.8Gbytes free.

Video clips are 2.02Gbytes

**Comment by someotherbob** [comment]
I updated to the newest daily build (12650).  After loading my project with 2 clips, Openshot crashes/closes.

[libopenshot.log](https://github.com/user-attachments/files/16155503/libopenshot.log)
[openshot-qt.log](https://github.com/user-attachments/files/16155505/openshot-qt.log)

**Comment by Colorjet3** [comment]
Hmmm...Did you try this option yet?

Option 5
1. Take your original files and run them through a converter like VLC, Handbrake, or ShutterEncoder and convert them to .mp4 even though the originals are .mp4. The conversion process will automatically create the new files with new names.

2. Now import these new converted files into a new project, perform your edtis, and now export.  Did this work?

**Update:**  Please provide the following:
1.  Computer specs:  CPU, GPU/RAM, RAM?
2.  What operating system/version are you running?

**Comment by someotherbob** [comment]
> Option 5
> 
> 1. Take your original files and run them through a converter like VLC, Handbrake, or ShutterEncoder and convert them to .mp4 even though the originals are .mp4. The conversion process will automatically create the new files with new names.
> 2. Now import these new converted files into a new project, perform your edtis, and now export.  Did this work?

What should I transcode to?  1080p 30 or 60?


> **Update:** Please provide the following:
> 
> 1. Computer specs:  CPU, GPU/RAM, RAM?

Intel(R) Celeron(R) J4105 CPU @ 1.50GHz   1.50 GHz
8.00 GB (7.84 GB usable)

> 2. What operating system/version are you running?

Win10 Enterprise, 22H2  All updates are applied.

**Comment by Colorjet3** [comment]
I would trans code to 1080p.

I think your computer is very under powered and in the end OpenShot may not be able to perform what you need.

**Comment by someotherbob** [comment]
> Option 5
> 
> 1. Take your original files and run them through a converter like VLC, Handbrake, or ShutterEncoder and convert them to .mp4 even though the originals are .mp4. The conversion process will automatically create the new files with new names.
> 2. Now import these new converted files into a new project, perform your edtis, and now export.  Did this work?
> 


I transcoded the videos and a project with all 12 clips renders at 1x.  I can then render that one video at 2x.

If I try to render the project with all 12 clips at 2x, the export fails.

**Comment by someotherbob** [comment]
> I would trans code to 1080p.
> 
> I think your computer is very under powered and in the end OpenShot may not be able to perform what you need.

If everything works, I am not concerned too much about render time.  I have not seen any memory pressure during a render, it never goes over 65% used.

**Comment by Colorjet3** [comment]
Hello @someotherbob.  This is beyond my expertise.  We need the lead developer to chime in.

**Comment by someotherbob** [comment]
Thanks for all the suggestions of how to debug.

I moved my media to a bigger machine (Core i7 with 64G memory).  I am not getting the stall at 6.36%, but Openshot hangs while moving clips around.  No consistent action causes this, seems to be random.

**Comment by Colorjet3** [comment]
Hello @someotherbob Version 3.2.1 was just release to production.  go to openshot.org, download and install this latest version.

I am going to close this ticket since it seems that the solution was to move to a more powerful computer.

If the "Openshot hangs while moving clips around" continues, then please submit a new thread so we can deal with that separately.

Just out of curiosity, when OpenShot hangs, does it eventually move the clips around or does OpenShot freeze?

**Comment by someotherbob** [comment]
> Hello @someotherbob Version 3.2.1 was just release to production. go to openshot.org, download and install this latest version.
> 
> I am going to close this ticket since it seems that the solution was to move to a more powerful computer.
> 
> If the "Openshot hangs while moving clips around" continues, then please submit a new thread so we can deal with that separately.
> 
> Just out of curiosity, when OpenShot hangs, does it eventually move the clips around or does OpenShot freeze?

Complete freeze, spinning blue circle cursor.  Sometimes the Openshot UI is greyed out.  I still cannot get a an export.

**Comment by Colorjet3** [comment]
I am still suspecting your under powered computer.  If you have access to something more powerful, I would try it there.

**Comment by someotherbob** [comment]
> I am still suspecting your under powered computer. If you have access to something more powerful, I would try it there.

I moved everything over to an Omen gaming system.  Core i7 and 64G RAM.

I can't even get to the point to try an export, Openshot freezes after a few edit actions on the timeline.

I suspected my external hard drive was the issue.  I moved the same 12 clips to drive C: and had the same freeze issue.

Thanks for all your work on this, but I've had to move on to a different video editor.  I'm out of debug time.
